src/api/box-edit/ComServerClient.js

```diff
@@ -0,0 +1,329 @@
+ *
+ * @param {number} [customTimeoutMS] optional field to override the timeout value passed in miliseconds
+ * @returns {number}
+ */
+function validateAndReturnBrowserToComServerTimeout(customTimeoutMS) {
+    let timeoutMS = REQUEST_TIMEOUT_MS;
+
+    // validate timeout is a positive number
+    if (typeof customTimeoutMS === 'number' && customTimeoutMS >= 0) {
+        timeoutMS = Math.floor(customTimeoutMS);
+    }
+
+    return timeoutMS;
+}
+
+/**
+ * Returns reduced timeout converted to seconds
+ * We need to use a shortened timeout for the connection between local com server and application,
+ * so that we will receive a message that that connection has timed out,
+ * before the connection between the browser and the local com server itself times out.
+ */
+function shortenAndReturnComServerToApplicationTimeout(browserToComServerTimeoutMS) {
+    let timeoutSec = +(browserToComServerTimeoutMS / 1000).toFixed(2);
+
+    if (browserToComServerTimeoutMS < 2) {
+        timeoutSec /= 2;
+    } else {
+        timeoutSec -= 1;
+    }
+
+    return timeoutSec;
+}
+
+function initBoxToolsLogData() {
+    const browserName = Browser.getName();
+    const browserVersion = Browser.getVersion();
+    const { UNCREATED_STATUS } = CONSTANTS;
+
+    boxToolsLogData = {
+        box_tools_version: null,
+        browser_name: browserName,
+        browser_version: browserVersion,
+        error_message: null,
+        installation_type: null,
+        http_channel_status: UNCREATED_STATUS,
+        https_channel_status: UNCREATED_STATUS,
+        activex_channel_status: UNCREATED_STATUS,
+        safari_channel_status: UNCREATED_STATUS,
+    };
+}
+
+class ComServerClient {
+    activeChannel: Channel;
+
+    channels: Array<Channel>;
+
+    isInitialized: boolean = false;
+
+    browser: string;
+
+    constructor(appName: string) {
+        this.channels = [];
+
+        this.isInitialized = true;
+        initBoxToolsLogData();
+
+        if (isMixedContentAllowedOnLocalhost()) {
+            this.channels.push(createHTTPChannel(appName));
+        } else if (isSupportedSafariVersion()) {
+            this.channels.push(createSafariChannel(appName));
+        } else if (isSupportedIEAndBoxToolsPluginAvailable()) {
+            this.channels.push(createActiveXChannel(appName));
+        } else if (isFirefoxWithExtensionsCapability() || isUnsupportedMSEdgeVersion()) {
+            // @NOTE (2018-07-24) No Action - Trying all channels is not an option in this case
+            // @TODO (2018-07-24) Remove this empty case from here?
+        } else {
+            // @NOTE: (2018-01-16) Trying all channels in case of custom useragent
+            this.channels = this.channels.concat([
+                createHTTPChannel(appName),
+                createSafariChannel(appName),
+                createSynchronousActiveXChannel(appName),
+            ]);
+        }
+    }
+
+    isInitialized() {
+        return this.isInitialized;
```

Comment:
This will just return the function property `isInitialized()`. Maybe rename this variable to something else